guided.py

```diff
@@ -0,0 +1,359 @@
+		if (passwd := archinstall.get_password(prompt='Enter disk encryption password (leave blank for no encryption): ')):
+			archinstall.arguments['!encryption-password'] = passwd
+			archinstall.arguments['harddrive'].encryption_password = archinstall.arguments['!encryption-password']
+
+	# Get the hostname for the machine
+	if not archinstall.arguments.get('hostname', None):
+		archinstall.arguments['hostname'] = input('Desired hostname for the installation: ').strip(' ')
+
+	# Ask for a root password (optional, but triggers requirement for super-user if skipped)
+	if not archinstall.arguments.get('!root-password', None):
+		archinstall.arguments['!root-password'] = archinstall.get_password(prompt='Enter root password (Recommendation: leave blank to leave root disabled): ')
+
+	# Ask for additional users (super-user if root pw was not set)
+	archinstall.arguments['users'] = {}
+	archinstall.arguments['superusers'] = {}
+	if not archinstall.arguments.get('!root-password', None):
+		archinstall.arguments['superusers'] = archinstall.ask_for_superuser_account('Create a required super-user with sudo privileges: ', forced=True)
+
+	users, superusers = archinstall.ask_for_additional_users('Enter a username to create a additional user (leave blank to skip & continue): ')
+	archinstall.arguments['users'] = users
+	archinstall.arguments['superusers'] = {**archinstall.arguments['superusers'], **superusers}
+
+	# Ask for archinstall-specific profiles (such as desktop environments etc)
+	if not archinstall.arguments.get('profile', None):
+		archinstall.arguments['profile'] = archinstall.select_profile(filter(lambda profile: (Profile(None, profile).is_top_level_profile()), archinstall.list_profiles()))
+	else:
+		archinstall.arguments['profile'] = archinstall.list_profiles()[archinstall.arguments['profile']]
+
+	# Check the potentially selected profiles preparations to get early checks if some additional questions are needed.
+	if archinstall.arguments['profile'] and archinstall.arguments['profile'].has_prep_function():
+		with archinstall.arguments['profile'].load_instructions(namespace=f"{archinstall.arguments['profile'].namespace}.py") as imported:
+			if not imported._prep_function():
+				archinstall.log(
+					' * Profile\'s preparation requirements was not fulfilled.',
+					fg='red'
+				)
+				exit(1)
+
+	# Ask about audio server selection if one is not already set
+	if not archinstall.arguments.get('audio', None):
+		
+		# only ask for audio server selection on a desktop profile 
+		if str(archinstall.arguments['profile']) == 'Profile(desktop)':
+			archinstall.arguments['audio'] = archinstall.ask_for_audio_selection()
+		else:
+			# packages installed by a profile may depend on audio and something may get installed anyways, not much we can do about that.
+			# we will not try to remove packages post-installation to not have audio, as that may cause multiple issues
+			archinstall.arguments['audio'] = None
+   
+   	# Ask what kernel user wants:
```

Comment:
Move the actual choice logic to the user_interaction library and call it as a function.

Reply:
Sorry, it shows me code I didn't even write:
```
			# we will not try to remove packages post-installation to not have audio, as that may cause multiple issues
			archinstall.arguments['audio'] = None

   	# Ask what kernel user wants:
```

what code are you referring to?

Reply:
The block from 180 onwards that has the user selecting a kernel. That choice would be best in user_interaction.py. it also should use the generic_select function

Reply:
> That choice would be best in user_interaction.py.

so do I put all of that code in `user_interaction.py` as a function and call it in guided.py?